### PR TITLE
debuging 1 outline

### PR DESCRIPTION
**Pull request by trainings-au**

### CODE REVIEW

File: outlines/all_outlines_foundations.md [hunk 1]
```diff
@@ -68,18 +79,26 @@ _Introducing a recipe for debugging code._
...
+    * "Rubber duck" explanations help you organize your thoughts.
+    * Believe in logical reasons, not magic -- logically connect cause and effect.
+
 
 ### Let's practice
 
-{exercises go here}
+#### Summative task:
+ * Debugging is early in the course - functions, lists, and maps are known.
+ * The example app is a calorie counter app e.g. https://github.com/davidhealey/waistline
```
Comment (manzoid) [inline outlines/all_outlines_foundations.md:96]: this seems way too hard doesn't it?  This is an Android app... how are they supposed to fix an Android app?

File: outlines/all_outlines_foundations.md [hunk 2]
```diff
@@ -55,6 +55,17 @@ _Reorganizing code to make it more testable._
...
 
+### Defense lines against bugs
+1. When we can detect a bug:
+   * end user
+   * testers
+   * reviewers
+   * autoamted tests
+   * programmers
+1. Why seasoned programmers create applications with less bugs?
+   * They follow clean code rules
```
Comment (manzoid) [inline outlines/all_outlines_foundations.md:66]: This isn't the only reason is it? 

For example, seasoned programmers also know to pay a lot of attention to knowing about the problem domain, they don't just think about code. They select dependable tools. They code review well, and accept code review feedback well.  They design ahead of time, in writing. They actively seek design reviews.  Etc, etc.

Better to caveat as "One important reason: they follow clean code rules."

File: outlines/all_outlines_foundations.md [hunk 3]
```diff
@@ -55,6 +55,17 @@ _Reorganizing code to make it more testable._
...
+   * You don't need years of expirience to follow them
```
Comment (manzoid) [inline outlines/all_outlines_foundations.md:67]: experience

File: outlines/all_outlines_foundations.md [hunk 4]
```diff
@@ -68,18 +79,26 @@ _Introducing a recipe for debugging code._
...
+1. Don't just look for quick workarounds, aim to understand what is happening.
+    * "Rubber duck" explanations help you organize your thoughts.
+    * Believe in logical reasons, not magic -- logically connect cause and effect.
+
 
 ### Let's practice
 
-{exercises go here}
+#### Summative task:
+ * Debugging is early in the course - functions, lists, and maps are known.
```
Comment (manzoid) [inline outlines/all_outlines_foundations.md:95]: Please list the points above that are being targeted by the Summative exercise.  

It shouldn't be _all_ of them, just some of them. 

We don't really need or particularly want to define the exact exercise and the exercise steps at the level of this outline.

**Review by manzoid — CHANGES_REQUESTED** [review]

File: outlines/all_outlines_foundations.md (hunk 1 above)
Comment (trainings-au) [reply outlines/all_outlines_foundations.md:96]: The idea is that this will be Python written by us. Just the business logic of the parts of the app.

**Comment by manzoid** [comment]
merging, will edit directly, thanks.

**Pull request merged**